ampd/src/handlers/sui_verify_worker_set.rs

```diff
@@ -0,0 +1,227 @@
+use cosmrs::cosmwasm::MsgExecuteContract;
+use error_stack::ResultExt;
+use serde::Deserialize;
+use sui_types::base_types::{SuiAddress, TransactionDigest};
+use tracing::{info, info_span};
+
+use axelar_wasm_std::voting::PollId;
+use connection_router::state::ID_SEPARATOR;
+use cosmwasm_std::HexBinary;
+use cosmwasm_std::Uint128;
+use events::{Error::EventTypeMismatch, Event};
+use events_derive::try_from;
+use voting_verifier::msg::ExecuteMsg;
+
+use crate::event_processor::EventHandler;
+use crate::handlers::errors::Error;
+use crate::queue::queued_broadcaster::BroadcasterClient;
+use crate::sui::json_rpc::SuiClient;
+use crate::sui::verifier::verify_worker_set;
+use crate::types::TMAddress;
+
+#[derive(Deserialize, Debug)]
+pub struct Operators {
+    pub weights_by_addresses: Vec<(HexBinary, Uint128)>,
+    pub threshold: Uint128,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct WorkerSetConfirmation {
+    pub tx_id: TransactionDigest,
+    pub event_index: u64,
+    pub operators: Operators,
+}
+
+#[derive(Deserialize, Debug)]
+#[try_from("wasm-worker_set_poll_started")]
+struct PollStartedEvent {
+    #[serde(rename = "_contract_address")]
+    contract_address: TMAddress,
+    poll_id: PollId,
+    source_gateway_address: SuiAddress,
+    worker_set: WorkerSetConfirmation,
+    participants: Vec<TMAddress>,
+}
+
+pub struct Handler<C, B>
+where
+    C: SuiClient + Send + Sync,
+    B: BroadcasterClient,
+{
+    worker: TMAddress,
+    voting_verifier: TMAddress,
+    rpc_client: C,
+    broadcast_client: B,
+}
+
+impl<C, B> Handler<C, B>
+where
+    C: SuiClient + Send + Sync,
+    B: BroadcasterClient,
+{
+    pub fn new(
+        worker: TMAddress,
+        voting_verifier: TMAddress,
+        rpc_client: C,
+        broadcast_client: B,
+    ) -> Self {
+        Self {
+            worker,
+            voting_verifier,
+            rpc_client,
+            broadcast_client,
+        }
+    }
+    async fn broadcast_votes(
+        &self,
+        poll_id: PollId,
+        votes: Vec<bool>,
+    ) -> error_stack::Result<(), Error> {
+        let msg = serde_json::to_vec(&ExecuteMsg::Vote { poll_id, votes })
+            .expect("vote msg should serialize");
+        let tx = MsgExecuteContract {
+            sender: self.worker.as_ref().clone(),
+            contract: self.voting_verifier.as_ref().clone(),
+            msg,
+            funds: vec![],
+        };
+
+        self.broadcast_client
+            .broadcast(tx)
+            .await
+            .change_context(Error::Broadcaster)
+    }
+}
+
+#[async_trait]
+impl<C, B> EventHandler for Handler<C, B>
+where
+    C: SuiClient + Send + Sync,
+    B: BroadcasterClient + Send + Sync,
+{
+    type Err = Error;
+
+    async fn handle(&self, event: &Event) -> error_stack::Result<(), Error> {
+        let PollStartedEvent {
+            contract_address,
+            poll_id,
+            source_gateway_address,
+            worker_set,
+            participants,
+            ..
```

Comment:
```suggestion
```
All fields are used, no?

Reply:
nah, the source chain name was not used.